CMakeLists.txt

```diff
@@ -16,495 +17,218 @@ cmake_minimum_required (VERSION 2.8.12)
-  endif()
+if(DEFINED EXTERNAL_LINKER_FLAGS)
+  foreach(FLAG ${EXTERNAL_LINKER_FLAGS})
+    set(LINKER_FLAGS_COMMON "${LINKER_FLAGS_COMMON} ${FLAG}")
+  endforeach()
+endif()
 
-  function(declare_target_with_modifiers) # modifiers are passed in ARGN implicit argument
-   set(CORE_TARGET_NAME ${BUILD_MODE_PREFIX_${BUILD_MODE}})
-   foreach(MODIFIER ${ARGN})
-    set(TARGET_NAME ${TARGET_NAME}${MODIFIER_SUFFIX_${MODIFIER}})
-
-    set(CORE_TARGET_NAME ${CORE_TARGET_NAME}${MODIFIER_SUFFIX_${MODIFIER}})
-   endforeach()
-   set(LIBM_TARGET_NAME ${CORE_TARGET_NAME}.jerry-libm.lib)
-   set(CORE_TARGET_NAME ${CORE_TARGET_NAME}.jerry-core)
-
-   set(DEFINES_JERRY )
-
-   if(NOT ${EXTERNAL_BUILD})
-    add_executable(${TARGET_NAME} ${SOURCE_JERRY_STANDALONE_MAIN})
-
-    if("${PLATFORM}" STREQUAL "MCU")
-     set(MCU_SCRIPT_GENERATED_HEADER ${MCU_SCRIPT_GENERATED_HEADER}.${TARGET_NAME})
-     add_custom_command(OUTPUT ${MCU_SCRIPT_GENERATED_HEADER}
-                        COMMAND ${CMAKE_SOURCE_DIR}/tools/generator.sh "${CMAKE_SOURCE_DIR}/${MCU_SCRIPT_FILE}" ${MCU_SCRIPT_GENERATED_HEADER}
-                        WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
-     add_custom_target(mcu_header_with_script_to_run.${TARGET_NAME} DEPENDS ${MCU_SCRIPT_GENERATED_HEADER})
-     set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_MCU_SCRIPT_HEADER="${MCU_SCRIPT_GENERATED_HEADER}")
-    elseif(("${PLATFORM}" STREQUAL "LINUX") OR ("${PLATFORM}" STREQUAL "DARWIN"))
-     if("${ENABLE_LOG}" STREQUAL "ON")
-      set(DEFINES_JERRY ${DEFINES_JERRY} JERRY_ENABLE_LOG)
-     endif()
-    endif()
+# Disable static build
+if(NOT ("${PLATFORM}" STREQUAL "DARWIN"))
+  set(LINKER_FLAGS_STATIC "-static")
+endif()
 
-    set_property(TARGET ${TARGET_NAME}
-                 PROPERTY COMPILE_FLAGS "${COMPILE_FLAGS_JERRY} ${FLAGS_COMMON_${BUILD_MODE}} ${C_FLAGS_JERRY}")
-    set_property(TARGET ${TARGET_NAME}
-                 PROPERTY LINK_FLAGS "${COMPILE_FLAGS_JERRY} ${FLAGS_COMMON_${BUILD_MODE}} ${LINKER_FLAGS_COMMON} ${LINKER_FLAGS_STATIC}")
-    target_compile_definitions(${TARGET_NAME} PRIVATE ${DEFINES_JERRY})
-    target_include_directories(${TARGET_NAME} PRIVATE ${INCLUDE_CORE_INTERFACE})
-    target_include_directories(${TARGET_NAME} PRIVATE ${PORT_DIR})
-    target_include_directories(${TARGET_NAME} SYSTEM PRIVATE ${INCLUDE_LIBC_INTERFACE})
-    target_include_directories(${TARGET_NAME} SYSTEM PRIVATE ${INCLUDE_EXTERNAL_LIBS_INTERFACE})
-    if(("${PLATFORM}" STREQUAL "DARWIN") AND (NOT (CMAKE_COMPILER_IS_GNUCC)))
-     target_link_libraries(${TARGET_NAME} ${CORE_TARGET_NAME} ${LIBC_TARGET_NAME}
-                           ${LIBM_TARGET_NAME} ${PREFIX_IMPORTED_LIB}libclang_rt.osx)
-    else()
-     target_link_libraries(${TARGET_NAME} ${CORE_TARGET_NAME} ${LIBC_TARGET_NAME}
-                           ${LIBM_TARGET_NAME} ${PREFIX_IMPORTED_LIB}libgcc)
-    endif()
+# C
+set(C_FLAGS_COMMON "-std=c99 -pedantic")
```

Comment:
It looks to me that `C_FLAGS_COMMON` and `COMPILE_FLAGS_COMMON` could be merged, and could probably even be replaced with CMAKE_C_FLAGS.